### PR TITLE
Drop PHP 7.1 Update to phpunit8 and add phpstan and phan

### PR DESCRIPTION
**Pull request by jasson99**
## Description:
This PR enables:
- [x] phpstan
- [x] phan
- [x] phpunit 8
 
## Related Issue: 
- https://github.com/owncloud/updater/issues/582

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -113,6 +115,16 @@ test-php-style-fix:        ## Run php-cs-fixer and fix code style issues
...
 
+
+.PHONY: test-php-phan
+test-php-phan: vendor-bin/phan/vendor
+	$(PHAN) --config-file .phan/config.php --require-config-exists -p
+
+
+.PHONY: test-php-phpstan
+test-php-phpstan: vendor-bin/phpstan/vendor
+	$(PHPSTAN) analyse --memory-limit=2G --configuration=./phpstan.neon --level=0 src/Tests
```
Comment (phil-davis) [inline Makefile:126]: Why are we only running `phpstan` on the `src/Tests` folder?

File: phpstan.neon [hunk 2]
```diff
@@ -0,0 +1,7 @@
+parameters:
+  bootstrapFiles:
+    - %currentWorkingDirectory%/app/bootstrap.php
+    - %currentWorkingDirectory%/app/config/container.php
+
+  excludes_analyse:
+    - %currentWorkingDirectory%/src/Tests/*/*.php
```
Comment (phil-davis) [inline phpstan.neon:7]: This excludes files, so that CI reports that only 2 files are analysed.
https://drone.owncloud.com/owncloud/updater/122/2/4

**Comment by phil-davis** [comment]
What is going on? `make test-php-phan` is passing locally. In GitHub this branch has the changes to files like `src/Utils/ZipExtractor.php`. https://drone.owncloud.com/owncloud/updater/128/3/1 shows that the changes are there in the clone step of the pipeline:
`src/Utils/ZipExtractor.php                       |  5 +-`
But `phan` https://drone.owncloud.com/owncloud/updater/128/3/3 reports problems like:
```
src/Utils/ZipExtractor.php:71 PhanDeprecatedFunction Call to deprecated function \Symfony\Component\Process\ProcessUtils::escapeArgument() defined at vendor/symfony/process/ProcessUtils.php:41 (Deprecated because: since version 3.3, to be removed in 4.0. Use a command line array or give env vars to the `Process::start/run()` method instead.)
```

But `src/Utils/ZipExtractor.php:71` is actually at line 72 now, and has `phan-suppress-next-line`

**Comment by VicDeo** [comment]
@phil-davis I spotted some kind of caching in other PR either
https://drone.owncloud.com/owncloud/updater/134/5/5
```
/var/www/owncloud/server/updater/src/Utils/Fetcher.php:193
```
But Fetcher:193 is a closing  PHPDoc line:
https://github.com/owncloud/updater/blob/f1cef38394e3265968a32a1fe8ec2517909c33b4/src/Utils/Fetcher.php#L193

**Comment by phil-davis** [comment]
I amended and force-pushed the last commit. Let's see what CI  does with it.

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
The "magic" was to put the "app" in `'path': 'server/apps/%s' % config['app']` (see the diffs of this PR)
Somehow, just having it in `server/updater` caused the issues with the analyzer etc not seeing the real files for this PR.

Actually the whole thing should be able to work without having an oC10 install. This `updater` is not an ordinary oC10 app.

After this PR is merged, I will try removing all the `installCore` ... stuff from the drone CI. That will avoid confusion for someone in future.

**Pull request merged**